pilot/pkg/model/context.go

```diff
@@ -849,14 +858,35 @@ func (node *Proxy) SetServiceInstances(serviceDiscovery ServiceDiscovery) {
 	node.ServiceInstances = instances
 }
 
-// SetWorkloadLabels will set the node.Metadata.Labels only when it is nil.
+// SetWorkloadLabels will set the node.Labels.
+// It merges both node meta labels and workload labels and give preference to workload labels.
 func (node *Proxy) SetWorkloadLabels(env *Environment) {
-	// First get the workload labels from node meta
-	if len(node.Metadata.Labels) > 0 {
+	if len(node.Labels) > 0 {
 		return
 	}
-	// Fallback to calling GetProxyWorkloadLabels
-	node.Metadata.Labels = env.GetProxyWorkloadLabels(node)
+	labels := env.GetProxyWorkloadLabels(node)
+	// when IstioMetaLabels unset, calculate the IstioMetaLabels. It is only set once.
```

Comment:
Do we want to do this as part of this PR?

Reply:
Maybe, to show usage